### DISCUSSION TITLE
Custom routes using rewrites don’t work on vercel dev

### DISCUSSION THREAD

**Discussion by hcentelles** [vercel/vercel#4538] [Help]
I’ve several rewrites defined on `vercel.json` for custom Next.js SSR pages routing, it works perfectly on deployments but not local with `vercel dev`.  Any idea how to make this work. 

Thanks 🙏

**Comment by TooTallNate** [comment]
Hey there @hcentelles. Rewrites should be working in `vercel dev`, but perhaps you're hitting an edge case this is not yet working properly. Could you give us a repository that we can repro this issue with to take a closer look? Thanks!

**Reply by hcentelles** [reply]
Sure @TooTallNate 

https://github.com/hcentelles/vercel-dev-rewrites-test

https://vercel-dev-rewrites-test.now.sh/

With `vercel dev` I get a 404 from that link.

Thanks

**Reply by ctrlplusb** [reply]
try popping a leading `/` in front of your `dest`. 👍

**Reply by hcentelles** [reply]
@ctrlplusb I've tried that without luck. It keeps showing a 404 error when using `vercel dev`

**Reply by hcentelles** [reply]
@TooTallNate any news about this?

Thanks

**Comment by davidknezic** [comment]
I think I have the same issue with the followign setup:

* one nextjs app
* one rewrite from `/http-bind` to external url

When deployed, the rewrite works:
https://now-nextjs-issues-git-vercel-rewrites.davidknezic.vercel.app/http-bind

When using `vercel dev`, I get the Vercel 404 page:
http://localhost:3000/http-bind

Simple rep(r)o:
https://github.com/davidknezic/now-nextjs-issues/tree/vercel-rewrites

Using:
Vercel CLI 19.1.2-canary.14 dev (beta)

**Comment by jakst** [comment]
I also have this problem when deploying nextjs apps with Vercel.

```
{
    "rewrites": [{ "source": "/rewrite", "destination": "https://example.com" }]
}
```

Adding this config to a freshly generated nextjs app (with `vc init` or `create-next-app`) and running `vc dev` will return the index.js page instead of rewriting when visiting `http://localhost:3000/rewrite`.

Vercel CLI version: both `19.1.2` and `19.1.3-canary.1`.
NextJS version: `9.4.4`

**Reply by jakst** [reply]
Git repo: https://github.com/Jakst/vercel-next-rewrites
Deployment: https://vercel-next-rewrites.vercel.app

**Reply by jakst** [reply]
I got a mail from support saying this was fixed, but it's still broken for me with the following versions:

vercel@19.2.1-canary.2
next@9.4.5-canary.43

**Reply by styfle** [reply]
Thanks, this will be fixed in PR #4914

**Reply by hcentelles** [reply]
Hi @styfle I've tried this with 19.2.1-canary.2 and 20.0.0-canary.14 but it's still broken for me too

**Reply by styfle** [reply]
I just tested with `vercel@20.0.0-canary.14` and it works fine.

```
npm i -g vercel@canary
git clone https://github.com/Jakst/vercel-next-rewrites
cd vercel-next-rewrites
yarn install
vc dev
# visit http://localhost:3000/rewrite
```

You could also try upgrading to Next.js 9.5 and defining `rewrites` in `next.config.js` instead.

See the announcement here: https://nextjs.org/blog/next-9-5#support-for-rewrites-redirects-and-headers

**Reply by hcentelles** [reply]
@styfle thanks for your answer. The repo you mentioned test the case of a rewrite with a external URL as destination, also the PR https://github.com/vercel/vercel/pull/4914 that you mentioned previously in this question as a fix, seems to deal with that case. 

What's broken for me is a rewrite with a serverless function as destination, as seen in the docs https://vercel.com/docs/configuration#project/rewrites (the 2nd and 3rd element of the list), in my case the serverless functions is a Next.js SSR page.

It works ok when deployed to Vercel but it doesn't with `vercel dev`.

Test repo: https://github.com/hcentelles/vercel-dev-rewrites-test

Vercel deployment: https://vercel-dev-rewrites-test.vercel.app/

**Reply by hcentelles** [reply]
@styfle I tried the rewrites feature of Next.js 9.5 and it works with `vc dev` and also deployed to Vercel. The same feature on `vercel.json` is still broken.
